tests/lit_processing/data_ingest/dqm_ingest/utils/test_md5sum_utils.py

```diff
@@ -1,103 +1,176 @@
         md5sum_PMID = md5sums[0]["md5sum"]
         assert md5sum_PMID == md5sum_data["PMID"]["PMID:9241"]
 
-        print("here to assert md5sum for FB")
-        md5sum_results = db.execute("select rmm.md5sum from cross_reference r, reference_mod_md5sum rmm, mod m where rmm.mod_id=m.mod_id and m.abbreviation='FB' and r.reference_id=rmm.reference_id and r.curie='FB:FBrf00000001' ")
-        md5sums = md5sum_results.fetchall()
+        # Assert md5sum for FB
+        md5sum_results = db.execute(text("""
+            SELECT rmm.md5sum
+            FROM cross_reference r
+            JOIN reference_mod_md5sum rmm ON r.reference_id = rmm.reference_id
+            JOIN mod m ON rmm.mod_id = m.mod_id
+            WHERE m.abbreviation='FB' AND r.curie='FB:FBrf00000001'
+        """))
+        md5sums = md5sum_results.mappings().fetchall()
+        assert md5sums, "No md5sum found for FB:FBrf00000001"
         md5sum_FB = md5sums[0]["md5sum"]
         assert md5sum_FB == md5sum_data["FB"]["FB:FBrf00000001"]
 
-        print("here to assert md5sum for XB")
-        md5sum_results = db.execute("select rmm.md5sum from cross_reference r, reference_mod_md5sum rmm, mod m where rmm.mod_id=m.mod_id and m.abbreviation='XB' and r.reference_id=rmm.reference_id and r.curie='Xenbase:XB-ART-58863' ")
-        md5sums = md5sum_results.fetchall()
+        # Assert md5sum for XB
+        md5sum_results = db.execute(text("""
+            SELECT rmm.md5sum
+            FROM cross_reference r
+            JOIN reference_mod_md5sum rmm ON r.reference_id = rmm.reference_id
+            JOIN mod m ON rmm.mod_id = m.mod_id
+            WHERE m.abbreviation='XB' AND r.curie='Xenbase:XB-ART-58863'
+        """))
+        md5sums = md5sum_results.mappings().fetchall()
+        assert md5sums, "No md5sum found for Xenbase:XB-ART-58863"
         md5sum_XB = md5sums[0]["md5sum"]
         assert md5sum_XB == md5sum_data["XB"]["Xenbase:XB-ART-58863"]
 
+        # Test with empty md5sum_data
         md5sum_data_empty = {}
         save_database_md5data(md5sum_data_empty)
-        md5sum_results = db.execute("select rmm.md5sum from cross_reference r, reference_mod_md5sum rmm, mod m where rmm.mod_id=m.mod_id and m.abbreviation='FB' and r.reference_id=rmm.reference_id and r.curie='FB:FBrf00000001' ")
-        md5sums = md5sum_results.fetchall()
-        md5sum_XB = md5sums[0]["md5sum"]
-        assert md5sum_XB == md5sum_data["FB"]["FB:FBrf00000001"]
-
+        # Assertions to verify no changes
+        md5sum_results = db.execute(text("""
+            SELECT rmm.md5sum
+            FROM cross_reference r
+            JOIN reference_mod_md5sum rmm ON r.reference_id = rmm.reference_id
+            JOIN mod m ON rmm.mod_id = m.mod_id
+            WHERE m.abbreviation='FB' AND r.curie='FB:FBrf00000001'
+        """))
+        md5sums = md5sum_results.mappings().fetchall()
+        assert md5sums, "No md5sum found for FB:FBrf00000001 after empty update"

```

Comment:
Does this assert that md5sums exists or isn't None or has some result ?

Reply:
Yes, this assert statement checks if md5sums exists, or it is not None, not empty, or not equivalent to False

Reply:
cool cool